ConsentViewController/Classes/SourcePointClient/SourcepointClientCoordinator.swift

```diff
@@ -386,73 +386,94 @@ class SourcepointClientCoordinator: SPClientCoordinator {
         from state: State,
         pubData: SPPublisherData?,
         messageMetaData: MessageMetaData?
-    ) -> PvDataRequestBody {
-        var ccpa: PvDataRequestBody.CCPA?
+    ) -> SPMobileCore.PvDataRequest {
+        var request: SPMobileCore.PvDataRequest?
         if let stateCCPA = state.ccpa {
-            ccpa = .init(
-                applies: stateCCPA.applies,
-                uuid: stateCCPA.uuid,
-                accountId: accountId,
-                siteId: propertyId,
-                consentStatus: stateCCPA.consentStatus,
-                pubData: pubData,
-                // swiftlint:disable:next force_unwrapping
-                messageId: messageMetaData != nil ? Int(messageMetaData!.messageId) : nil,
-                sampleRate: state.ccpaMetaData?.sampleRate
+            let pubDataJson = pubData?.toCore()
+            request = SPMobileCore.PvDataRequest.init(
+                gdpr: nil,
+                ccpa: SPMobileCore.PvDataRequest.CCPA.init(
+                    applies: stateCCPA.applies,
+                    uuid: stateCCPA.uuid,
+                    accountId: Int32(accountId),
+                    siteId: Int32(propertyId),
+                    consentStatus: stateCCPA.consentStatus.toCore(rejectedVendors: stateCCPA.rejectedVendors, rejectedCategories: stateCCPA.rejectedCategories),
+                    // swiftlint:disable:next force_unwrapping
+                    pubData: pubDataJson != nil ? JsonKt.StringToJsonObject(str: pubDataJson!) : nil,
```

Comment:
I think it's more practical if you refactor `StringToJsonObject` to receive an optional String and return nil if the string is optional. This way you don't have to do this ternary conditions here.